File: Week 2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string t = get_string("Text: ");
    int s = 0, l = 0, w = 1;

    // Count letters, words, sentences
    for (int i = 0, n = strlen(t); i < n; i++)
    {
        if (t[i] == ' ')
        {
            w++;
        }
        else if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            s++;
        }
        else if (isalpha(t[i]))
        {
            l++;
        }
    }

    // Calculate L and S
    float L = (float) l / w * 100;
    float S = (float) s / w * 100;

    // Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Output grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

File: week 6/readability.py
from cs50 import get_string

s = 0  # sentences
l = 0  # letters
w = 1  # words (start with 1 since last word has no space)

text = get_string("Text: ")
length = len(text)

for i in range(length):
    if text[i] == ' ':
        w += 1
    elif text[i] in ['.', '!', '?']:
        s += 1
    elif text[i].isalpha():   # FIXED here
        l += 1

# Coleman–Liau index formula
L = (l / w) * 100
S = (s / w) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Output result
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")  # FIXED here
